sdk/monitor/opentelemetry-exporter/src/platform/nodejs/persist/fileSystemPersist.ts

```diff
@@ -28,120 +35,99 @@ export class FileSystemPersist implements PersistentStorage {
+          const payload = await readFileAsync(filePath);
+          // delete the file first to prevent double sending
+          await unlinkAsync(filePath);
+          return payload;
         }
-      } else if (stats.isDirectory()) {
-        fs.readdir(tempDir, (error, origFiles) => {
-          if (!error) {
-            const files = origFiles.filter((f) =>
-              path.basename(f).includes(FileSystemPersist.FILENAME_SUFFIX)
-            );
-            if (files.length === 0) {
-              callback(null);
-            } else {
-              const firstFile = files[0];
-              const filePath = path.join(tempDir, firstFile);
-              fs.readFile(filePath, (readFileErr, payload) => {
-                if (!readFileErr) {
-                  // delete the file first to prevent double sending
-                  fs.unlink(filePath, (unlinkError) => {
-                    if (!unlinkError) {
-                      callback(null, payload);
-                    } else {
-                      callback(unlinkError);
-                    }
-                  });
-                } else {
-                  callback(readFileErr);
-                }
-              });
-            }
-          } else {
-            callback(error);
-          }
-        });
       }
-    });
+      return null;
+    } catch (e) {
+      if (((e as unknown) as { code: string }).code === "ENOENT") {
+        // File does not exist -- return null instead of throwing
+        return null;
+      } else {
+        throw e;
+      }
+    }
   }
 
-  private _storeToDisk(
-    payload: string,
-    cb: (error: Error | null, success?: boolean) => void
-  ): void {
+  private async _storeToDisk(payload: string): Promise<boolean> {
     const directory = path.join(
       os.tmpdir(),
       FileSystemPersist.TEMPDIR_PREFIX + this._options.instrumentationKey
     );
 
-    confirmDirExists(directory, (error) => {
-      if (error) {
-        this._logger.warn(`Error while checking/creating directory: `, error && error.message);
-        cb(error);
-        return;
-      }
+    try {
+      await confirmDirExists(directory);
+    } catch (error) {
+      this._logger.warn(`Error while checking/creating directory: `, error && error.message);
+      throw error;
+    }
 
-      getShallowDirectorySize(directory, (err, size) => {
-        if (err || size < 0) {
-          this._logger.warn(
-            `Error while checking directory size: ${err?.message ?? "unknown error"}`
-          );
-          cb(err);
-          return;
-        }
-        if (size > this.maxBytesOnDisk) {
-          this._logger.warn(
-            `Not saving data due to max size limit being met. Directory size in bytes is: ${size}`
-          );
-          cb(new Error(`FileSystemPersist size limit reached: ${this.maxBytesOnDisk}`));
-          return;
-        }
+    const size = await getShallowDirectorySize(directory);
+    if (size < 0) {
```

Comment:
I don't see where `getShallowDirectorySize ` would return a negative value in the new code. 

It feels like there should be a try/catch for errors here, since there isn't in `getShallowDirectorySize` ? Maybe alternatively `getShallowDirectorySize ` can try/catch and return -1 again?

Reply:
added try/catch here